magic.js

```diff
@@ -168,15 +157,18 @@ function rsaverify(digest, padding) {
    */
   function keying(key) {
     return new Promise((resolve, reject) => {
-      if (key.startsWith('-----BEGIN PUBLIC KEY-----'))     { return resolve(key); }
```

Comment:
PEM SPKI public keys are handled through `createPublicKey()` of which we then check the `asymmetricKeyType` to be `rsa`. `BEGIN PUBLIC KEY` label could just as well be an EC key which would then be used for ECDSA instead of RSASSA-PSS.